### PR TITLE
Allow to access tekton-results

### PR DESCRIPTION
**Pull request by Michkov**
Allow users to access results and records stored in tekton-results.

Tekton results: https://github.com/tektoncd/results
Deployed to stage cluster by https://github.com/redhat-appstudio/infra-deployments/pull/136

e2e tests: https://github.com/codeready-toolchain/toolchain-e2e/pull/500

### CODE REVIEW

**Comment by Michkov** [comment]
cc @sbose78 @goldmann

**Comment by goldmann** [comment]
This is great!

File: deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml [hunk 1]
```diff
@@ -68,6 +68,14 @@ objects:
...
     - create
     - delete
+  - apiGroups:
+    - results.tekton.dev
+    resources:
+    - results
+    - records
+    verbs:
+    - get
+    - list
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml:78]: I guess you want `watch` too?
```suggestion
    - list
    - watch
```

**Review by alexeykazakov — COMMENTED** [review]
Looks good but please:
- rebase the PR
- add a e2e test for this new rule here: https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/tiers/checks.go#L1015
You will need to create a PR for the e2e repo from your fork but with the same branch name `allow-tekton-results` as this PR, so our CI can pair those two PRs.

Thanks!

File: deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml (hunk 1 above)
Comment (Michkov) [reply deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml:78]: Not needed, it's not used in their clusterrole -> https://github.com/tektoncd/results/blob/main/config/default-clusterroles.yaml

**Comment by MatousJobanek** [comment]
@Michkov You either need to rebase the PR or give us permission to commit to your branch. If the PR is not up-to-date with master, then the e2e tests will not work.

File: deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml (hunk 1 above)
Comment (Michkov) [reply deploy/templates/nstemplatetiers/appstudio/ns_appstudio.yaml:85]: This is not a real resource, but it's used by tekton results api to use RBAC. https://github.com/tektoncd/results/tree/main/docs/api

**Review by alexeykazakov — APPROVED** [review]

**Pull request merged**